quantities/generators_body.hpp

```diff
@@ -26,30 +26,26 @@ struct Collapse : not_constructible {
   using Type = Q;
 };
 
-template<>
+template<template<typename> typename Quantity>
```

Comment:
Put a comment at the beginning of this file explaining that the template template parameter `Quantity` lifts a circular dependency.